numpy/lib/npyio.py

```diff
@@ -86,10 +86,17 @@ def __dir__(self):
         return object.__getattribute__(self, '_obj').keys()
 
 
-def zipfile_factory(*args, **kwargs):
+def zipfile_factory(file, *args, **kwargs):
```

Comment:
This is not changing the effective signature - `*args` was being passed to `zipfile.ZipFile` before, and `zipfile.ZipFile` takes `file` as its first argument, so anything that worked before should continue to work.